assembl/static2/js/app/components/administration/landingPage/manageModules.jsx

```diff
@@ -16,16 +18,80 @@ import {
 } from '../../../actions/adminActions/landingPage';
 import { createRandomId } from '../../../utils/globalFunctions';
 
+export type LandingPageModuleType = {
+  defaultOrder: number,
+  editableOrder: boolean,
+  id: string,
+  identifier: string,
+  moduleId: string,
+  required: boolean,
+  title: string
+};
+
+type LandingPageModule = {
+  configuration: Object,
+  enabled: boolean,
+  existsInDatabase: true,
+  id: string,
+  moduleType: LandingPageModule,
+  order: number,
+  subtitle: ?string,
+  subtitleEntries: Array<LangstringEntries>,
+  title: ?string,
+  titleEntries: Array<LangstringEntries>
+};
+
 type Props = {
   enabledModules: List<Map>,
-  moduleTypes: Array<Object>,
+  moduleTypes: Array<LandingPageModuleType>,
   locale: string,
   modulesById: Map<string, Map>,
   moveModuleDown: Function,
   moveModuleUp: Function,
   toggleModule: Function,
   createModule: Function
 };
+
+const MODULES_IDENTIFIERS = {
+  header: 'HEADER',
+  timeline: 'TIMELINE',
+  chatbot: 'CHATBOT',
+  topThematics: 'TOP_THEMATICS',
+  tweets: 'TWEETS',
+  contact: 'CONTACT',
+  data: 'DATA',
+  news: 'NEWS',
+  partners: 'PARTNERS',
+  introduction: 'INTRODUCTION',
+  footer: 'FOOTER'
+};
+
+export const addEnumSuffixToModuleTitles = (modules: Array<LandingPageModule>): Array<LandingPageModule> => {
```

Comment:
you don't need this function here, it should move to the file in which it is used or in a distinct file (maybe in `utils/` folder but it is probably overkill and not very easy to make it clear and reusable)

Reply:
Well I can put it in utils because it's used in administration.jsx and this file doesn't have flow nor a test file. Of course we need to add those 2 things but it's not the purpose of this branch. Anyway administration.jsx is already too big